### PR TITLE
Bundling scripts

### PR DESCRIPTION
**Pull request by isaacroldan**
### WHY are these changes introduced?
As part of the Global CLI project we are also introducing dependency bundling. This PR adds all the necessary scripts to bundle the CLI.

### WHAT is this pull request doing?
- Add bundle scripts for `cli` and `create-app`.
- Add some utilities shared by both bundle scripts.
- Add new `bundle` script in package.json and project.json for both packages.

THIS PR ONLY ADDS THE SCRIPTS, the bundling is not being triggered yet anywhere,

### How to test your changes?
Nothing to test.

### Measuring impact

How do we know this change was effective? Please choose one:

- [ ] n/a - this doesn't need measurement, e.g. a linting rule or a bug-fix
- [ ] Existing analytics will cater for this addition
- [ ] PR includes analytics changes to measure impact

### Checklist

- [ ] I've considered possible cross-platform impacts (Mac, Linux, Windows)
- [ ] I've considered possible [documentation](https://shopify.dev) changes
- [ ] I've made sure that any changes to `dev` or `deploy` have been reflected in the [internal flowchart](https://www.figma.com/file/7vqUp50u6dm48Zfb4JRRn8/CLI3-Internals).

### CODE REVIEW

File: packages/cli/bin/bundle.js [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+        {
+          from: ['../cli-kit/assets/**/*'],
+          to: ['./dist/assets'],
+        },
+        {
+          from: [yogafile],
+          to: ['./dist/'],
+        },
+        {
+          from: ['../../node_modules/.pnpm/node_modules/@shopify/theme-check-node/configs/*.yml'],
```
Comment (shauns) [inline packages/cli/bin/bundle.js:65]: This one feels a bit fragile -- any other ways we could do this? We can change `theme-check-node` if we need to

File: packages/create-app/bin/bundle.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+const external = [
+  // react-devtools-core is a dev dependency, no need to bundle it but throws errors if not included here.
+  'react-devtools-core',
+]
+
+// yoga wasm file is not bundled by esbuild, so we need to copy it manually
+const yogafile = glob.sync('../../node_modules/.pnpm/**/yoga.wasm')[0]
+
+esBuild({
```
Comment (shauns) [inline packages/create-app/bin/bundle.js:15]: If I change the big CLI bundle script, I really need to change this one too (e.g. maybe we realise some extra plugin is needed)

I wonder if there is a way to organise these to minimise the chance of missing one of them?

File: packages/cli/bin/bundle.js [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+  // react-devtools-core is a dev dependency, no need to bundle it but throws errors if not included here.
+  'react-devtools-core',
+  // esbuild can't be bundled per design
+  'esbuild',
+]
+
+// yoga wasm file is not bundled by esbuild, so we need to copy it manually
+const yogafile = glob.sync('../../node_modules/.pnpm/**/yoga.wasm')[0]
+
+esBuild({
```
Comment (shauns) [inline packages/cli/bin/bundle.js:21]: I guess a subsequent PR will add a GH action that triggers (and checks??) bundling?

File: packages/create-app/bin/bundle.js (hunk 2 above)
Comment (isaacroldan) [reply packages/create-app/bin/bundle.js:15]: If we added a shared script, we will still need specific config for each package (extra plugins, defines, external dependencies) so we end up with 3 script files and that is more risk of editing the wrong one maybe?

Not sure though. The create-app one just needs to solve stuff that comes from `cli-kit` (it's only dependency)

File: packages/cli/bin/bundle.js (hunk 3 above)
Comment (isaacroldan) [reply packages/cli/bin/bundle.js:21]: A subsequent script will add the shipit config to trigger the bundling.
I didn't add a GH action to validate that bundling works but will do :)

**Pull request merged**